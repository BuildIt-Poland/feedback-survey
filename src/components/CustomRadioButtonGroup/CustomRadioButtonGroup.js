import React from 'react';
import styled from 'styled-components';
import { bool, string, arrayOf } from 'prop-types';

import CustomRadioButton from '../CustomRadioButton';
import media from '../../styles/media';
import { radioOption } from '../../types';

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;

  ${media.mobileL`
    justify-content: space-evenly;
  `};
`;

const renderRadioButtons = (options, name, withIcon) =>
  options.map(radio => <CustomRadioButton key={radio.id} name={name} radio={radio} withIcon={withIcon} />);

const CustomRadioButtonGroup = ({ options, name, withIcon = false, onChange }) => (
  <Wrapper onChange={onChange}>{renderRadioButtons(options, name, withIcon)}</Wrapper>
);

CustomRadioButtonGroup.propTypes = {
  options: arrayOf(radioOption).isRequired,
  name: string.isRequired,
  withIcon: bool
};

export default CustomRadioButtonGroup;
