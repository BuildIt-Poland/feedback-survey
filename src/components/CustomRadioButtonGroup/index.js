import CustomRadioButtonGroup from './CustomRadioButtonGroup';

export default CustomRadioButtonGroup;
