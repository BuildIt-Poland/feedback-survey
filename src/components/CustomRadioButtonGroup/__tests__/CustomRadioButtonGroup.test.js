import React from 'react';
import { create } from 'react-test-renderer';

import CustomRadioButtonGroup from '..';

describe('COMPONENT - CustomRadioButtonGroup', () => {
  it('renders correctly', () => {
    const component = create(<CustomRadioButtonGroup />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
