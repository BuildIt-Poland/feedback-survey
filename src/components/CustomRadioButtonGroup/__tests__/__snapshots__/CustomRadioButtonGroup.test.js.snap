// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - CustomRadioButtonGroup renders correctly 1`] = `
<div
  className="sc-bxivhb cSJGRK"
>
  <input
    className="sc-bdVaJa kRURFE"
    id="test1"
    name="test"
    type="radio"
    value="test"
  />
  <label
    className="sc-htpNat OeGne"
    htmlFor="test1"
  >
    label-test
  </label>
  <input
    className="sc-bdVaJa kRURFE"
    id="test2"
    name="test"
    type="radio"
    value="test"
  />
  <label
    className="sc-htpNat OeGne"
    htmlFor="test2"
  >
    label-test
  </label>
</div>
`;

exports[`COMPONENT - CustomRadioButtonGroup renders correctly with icon 1`] = `
<div
  className="sc-bxivhb cSJGRK"
>
  <input
    className="sc-bdVaJa kRURFE"
    id="test1"
    name="test"
    type="radio"
    value="test"
  />
  <label
    className="sc-bwzfXH hTdoZH"
    htmlFor="test1"
  >
    <svg
      height="75"
      viewBox="0 0 490 490"
      width="75"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <g>
        <circle
          cx="157.651"
          cy="175.288"
          r="42.642"
        />
        <circle
          cx="332.349"
          cy="175.289"
          r="42.642"
        />
        <path
          d="M69.086,490h351.829C459.001,490,490,459.001,490,420.914V69.086C490,30.999,459.001,0,420.914,0H69.086 C30.991,0,0,30.999,0,69.086v351.829C0,459.001,30.991,490,69.086,490z M30.625,69.086c0-21.204,17.257-38.461,38.461-38.461 h351.829c21.204,0,38.461,17.257,38.461,38.461v351.829c0,21.204-17.257,38.461-38.461,38.461H69.086 c-21.204,0-38.461-17.257-38.461-38.461V69.086z"
        />
      </g>
    </svg>
    <span>
      label-test
    </span>
  </label>
  <input
    className="sc-bdVaJa kRURFE"
    id="test2"
    name="test"
    type="radio"
    value="test"
  />
  <label
    className="sc-bwzfXH hTdoZH"
    htmlFor="test2"
  >
    <svg
      height="75"
      viewBox="0 0 490 490"
      width="75"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <g>
        <circle
          cx="157.651"
          cy="175.288"
          r="42.642"
        />
        <circle
          cx="332.349"
          cy="175.289"
          r="42.642"
        />
        <path
          d="M69.086,490h351.829C459.001,490,490,459.001,490,420.914V69.086C490,30.999,459.001,0,420.914,0H69.086 C30.991,0,0,30.999,0,69.086v351.829C0,459.001,30.991,490,69.086,490z M30.625,69.086c0-21.204,17.257-38.461,38.461-38.461 h351.829c21.204,0,38.461,17.257,38.461,38.461v351.829c0,21.204-17.257,38.461-38.461,38.461H69.086 c-21.204,0-38.461-17.257-38.461-38.461V69.086z"
        />
      </g>
    </svg>
    <span>
      label-test
    </span>
  </label>
</div>
`;
